### ISSUE TITLE
Bento check fails due to "Argument list too long" error

### ISSUE THREAD

**Issue by dlukeomalley** [returntocorp/bento#253]
**EDIT BY @dlukeomalley**

To bypass this issue, please run on a subset of directories or files in your project: 

```bash
$ bento check --all [PATHS]
```

---

## Overview
Bento check fails to run, returning "argument list too long".

## Current Behavior
> Looks like you're handing a list off to some bash somewhere and that's
ticking over the bash max arg limit 

From console:
```bash
$ bento check

Running Bento checks...

r2c.bandit․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․|■■■■■■■□□□| 02:31, 🍤 Running
r2c.flake8․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․|■■■■■□□□□□| 02:31, 🍤 Running
r2c.flask․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․|■■■■■■■□□□| 02:31, 🍤 Running
r2c.requests․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․|■■□□□□□□□□| 02:31, 🍤 Running

✘ Error while running r2c.bandit: [Errno 7] Argument list too long:
'/bin/bash'
```

From last.log:
```bash
[ERROR] 215277.22716331482 root:check - [Errno 7] Argument list too
long: '/bin/bash'
[ERROR] 215277.48036384583 root:util - Error while running r2c.bandit:
[Errno 7] Argument list too long: '/bin/bash'
```

## Possible Solution
A better error message might help here (“please run on a subset of files”), along with removing the “try --clean” which doesn’t do anything in this situation.

## Additional Context
Bento version (run `bento --version`): 0.7.0+
Operation system (e.g. Ubuntu 18.04): Mac OSX

**Comment by dlukeomalley** [comment]
Additional users on 0.8.0 have also experienced this issue for large project.

**Comment by ievans** [comment]
If the bento collected list exceeds ARGMAX, we will see the following error:
```
[ERROR] 146361.19174957275 root:tool_runner - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/bento/tool_runner.py", line 261, in _setup_and_run_single_tool
    results = self._run_single_tool(bar, ix, tool, baseline)
  File "/usr/local/lib/python3.7/site-packages/bento/tool_runner.py", line 230, in _run_single_tool
    tool.tool_id(), tool.results(self.paths, self.use_cache), baseline
  File "/usr/local/lib/python3.7/site-packages/bento/tool.py", line 252, in results
    raw = self.run([str(p) for p in paths_to_run])
  File "/usr/local/lib/python3.7/site-packages/bento/extra/flake8.py", line 197, in run
    return self.venv_exec(cmd, check_output=False)
  File "/usr/local/lib/python3.7/site-packages/bento/extra/python_tool.py", line 79, in venv_exec
    stderr=subprocess.PIPE,
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 7] Argument list too long: 'python'
```

Another workaround: if the user is in linux, `ulimit -s 65536` will change ARG_MAX as it is now 1/4 of the stack size.

**Comment by dlukeomalley** [comment]
This issue has been addressed in v0.9, shipping later this week.
